src/gc-gpu-runner/CMakeLists.txt

```
@@ -15,7 +15,7 @@
 # SPDX-License-Identifier: Apache-2.0
 ################################################################################
 
-if(NOT GC_ENABLE_TOOLS OR NOT GC_ENABLE_IMEX)
+if(NOT GC_ENABLE_TOOLS OR NOT GC_ENABLE_GPU)
```

```suggestion
if(NOT GC_ENABLE_TOOLS AND NOT GC_ENABLE_GPU AND NOT GC_ENABLE_IMEX)
```
Is it correct or I have missed something?

---

In this PR, `GC_ENABLE_IMEX` is decoupled from the general GPU tools and is used to enable IMEX-specific passes (`GcGpuIMEXPasses`) and other parts of code that rely on IMEX-specific representations.
Actually, I would assume we should not build `gpu-runner` if the tools flag is explicitly set to false. For this particular case, I followed the existing convention of allowing `gpu-runner` to be built even when we have tools disabled. What was the original intent?

---

You are right, it shouldn't be built, if the tools flag is false, but it also shouldn't be built, if neither gpu nor imex are enabled, otherwise the build will fail, I assume.

---

Then the current flag is correct: we do not build `gpu-runner` if we do not want to (tools flag is false) OR we cannot (GPU is false). This PR enables GPU-related stuff to run both with and without setting `GC_ENABLE_IMEX`, so it is no longer a pre-requisite for `gpu-runner` to work.

---

Will it work properly if tools is true, but imex and gpu are false? If yes, then it's ok.

---

Current logic:
```
if(NOT GC_ENABLE_TOOLS OR NOT GC_ENABLE_GPU)
  message(STATUS "Gpu runner is not enabled.")
  return()
endif()
```
Possible cases are:
`GC_ENABLE_TOOLS = FALSE, GC_ENABLE_GPU = FALSE` :
We do not build, because the both conditions fire.
`GC_ENABLE_TOOLS = TRUE, GC_ENABLE_GPU = FALSE `:
We do not build, because the second condition fires.
`GC_ENABLE_TOOLS = FALSE, GC_ENABLE_GPU = TRUE` :
We do not build, because the first condition fires.
`GC_ENABLE_TOOLS = TRUE, GC_ENABLE_GPU = TRUE` :
We do build, because we want AND can.

I think we can avoid possible confusion by simplifying this to:
```
if(GC_ENABLE_TOOLS AND GC_ENABLE_GPU)
  gc_add_mlir_tool(gc-gpu-runner GpuRunner.cpp)
  target_link_libraries(gc-gpu-runner PRIVATE
    GcJitWrapper
    GcGpuOclRuntime
  )
  mlir_check_all_link_libraries(gc-gpu-runner)
else
  message(STATUS "Gpu runner is not enabled.")
  return()
endif()
```

---

This will not work if GC_ENABLE_TOOLS=on and GC_ENABLE_IMEX=on.